xfer/publisher.go

```diff
@@ -52,21 +53,30 @@ func NewHTTPPublisher(target, token, id string) (*HTTPPublisher, error) {
 
 // Publish publishes the report to the URL.
 func (p HTTPPublisher) Publish(rpt report.Report) error {
-	var buf bytes.Buffer
-	if err := gob.NewEncoder(&buf).Encode(rpt); err != nil {
+	gzbuf := &bytes.Buffer{}
```

Comment:
Minor comment: I prefer gzbuf := bytes.Buffer{} and then using the & operator when passing it to gzip.NewWriter and http.NewRequest